modules/apigee_edge_apidocs/src/Entity/ApiDoc.php

```diff
@@ -47,30 +51,43 @@
  *     "access" = "Drupal\apigee_edge_apidocs\ApiDocAccessControlHandler",
  *     "route_provider" = {
  *       "html" = "Drupal\apigee_edge_apidocs\ApiDocHtmlRouteProvider",
+ *       "revision" = "Drupal\entity\Routing\RevisionRouteProvider",
  *     },
  *   },
  *   base_table = "apidoc",
  *   data_table = "apidoc_field_data",
+ *   revision_table = "apidoc_revision",
+ *   revision_data_table = "apidoc_field_revision",
+ *   show_revision_ui = TRUE,
  *   translatable = TRUE,
  *   admin_permission = "administer apidoc entities",
  *   entity_keys = {
  *     "id" = "id",
  *     "label" = "name",
  *     "uuid" = "uuid",
  *     "langcode" = "langcode",
- *     "status" = "status",
+ *     "published" = "status",
+ *     "revision" = "revision_id",
+ *   },
+ *   revision_metadata_keys = {
+ *     "revision_user" = "revision_user",
+ *     "revision_created" = "revision_created",
+ *     "revision_log_message" = "revision_log_message",
  *   },
  *   links = {
  *     "canonical" = "/apidoc/{apidoc}",
  *     "add-form" = "/admin/structure/apidoc/add",
  *     "edit-form" = "/admin/structure/apidoc/{apidoc}/edit",
  *     "delete-form" = "/admin/structure/apidoc/{apidoc}/delete",
+ *     "version-history" = "/admin/structure/apidoc/{apidoc}/revisions",
+ *     "revision" = "/admin/structure/apidoc/{apidoc}/revisions/{apidoc_revision}/view",
+ *     "revision-revert-form" = "/admin/structure/apidoc/{apidoc}/revisions/{apidoc_revision}/revert",
  *     "collection" = "/admin/structure/apidoc",
  *   },
  *   field_ui_base_route = "apigee_edge_apidocs.settings"
  * )
  */
-class ApiDoc extends ContentEntityBase implements ApiDocInterface {
+class ApiDoc extends EditorialContentEntityBase implements ApiDocInterface {
```

Comment:
After I saw copy-pasting from `\Drupal\node\Entity\Node` several places here I need to ask, is there any reason why this class does not extend Node? That way even the node access would work on the API documentation (content) entity type, which is a requirement if we would like to support any kind of search module.